vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
+    "NUGET": NugetVersionRange,
+    "COMPOSER": ComposerVersionRange,
+    "PIP": PypiVersionRange,
+    "RUBYGEMS": GemVersionRange,
+    "GO": GolangVersionRange,
+}
+
+PKG_TYPE_ECOSYSTEM_MAPPING = {value: key for (key, value) in ECOSYSTEM_PKG_TYPE_MAPPING.items()}
+
+QUERY = """
+        query{
+        securityVulnerabilities(first: 100, ecosystem: %s, %s) {
+            edges {
+            node {
+                advisory {
+                identifiers {
+                    type
+                    value
+                }
+                summary
+                references {
+                    url
+                }
+                severity
+                publishedAt
+                }
+                package {
+                name
+                }
+                vulnerableVersionRange
+            }
+            }
+            pageInfo {
+            hasNextPage
+            endCursor
+            }
+        }
+        }
+        """
+
 
 class GitHubTokenError(Exception):
     pass
 
 
+def get_item(json_object: dict, *attributes):
+    """
+    Return `item` by going through all the `attributes` present in the `json_object`
+    """
+    if not json_object:
+        LOGGER.error(f"json_object is empty: {json_object}")
+        return None
+    item = json_object
+    for attribute in attributes:
+        if not attribute in item:
+            LOGGER.error(f"Missing attribute {attribute} in {item}")
+            return None
+        item = item[attribute]
+    return item
+
+
 class GitHubAPIImporter(Importer):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    spdx_license_expression = "CC-BY-4.0"
+    token = os.environ["GH_TOKEN"]
+    endpoint = "https://api.github.com/graphql"
+
+    ecosystems = [
+        "MAVEN",
+        "NUGET",
+        "COMPOSER",
+        "PIP",
+        "RUBYGEMS",
+        "GO",
+    ]
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        headers = {"Authorization": f"token {self.token}"}
+        end_cursor_exp = ""
+        advisories = []
+        for ecosystem in self.ecosystems:
+            while True:
+                query_json = {"query": QUERY % (ecosystem, end_cursor_exp)}
+                resp = requests.post(self.endpoint, headers=headers, json=query_json).json()
+                if resp.get("message") == "Bad credentials":
+                    raise GitHubTokenError("Invalid GitHub token")
+                advisories.extend(process_response(resp, ecosystem=ecosystem))
+                page_info = get_item(resp, "data", "securityVulnerabilities", "pageInfo")
+                if not get_item(page_info, "hasNextPage"):
+                    break
+                end_cursor = get_item(page_info, "endCursor")
+                if end_cursor:
+                    end_cursor = f'"{end_cursor}"'
+                    end_cursor_exp = f"after: {end_cursor}"
+        return advisories
+
+
+def get_ecosystem_from_type(type: str):
```

Comment:
Do we need a function for this, as opposed to make a direct get call?
Same for def get_type_from_ecosystem ... it feels too much